### PR TITLE
AT-5997 Resolve intermittent unit test failures regarding list comparisons

### PR DESCRIPTION
**Pull request by anhtran-ccpo**
Added method for sorting nested lists and resolve intermittently failing tests around list comparisons

Ticket [AT-5997](https://ccpo.atlassian.net/browse/AT-5997)

### CODE REVIEW

File: tests/models/test_application_role.py [hunk 1]
```diff
@@ -46,13 +46,10 @@ def test_has_application_role_history(session):
 
 def test_environment_roles():
     application = ApplicationFactory.create()
-    environment1 = EnvironmentFactory.create(application=application)
```
[the remark this answers is not in the document]
Comment (anhtran-ccpo) [reply tests/models/test_application_role.py]: Undo change

File: atat/utils/utilities.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def sort_nested(lst: List[list]):
+    """
+    This method is for sorting a list of lists. This as it is will only work with one depth or rather a matrix
+    The contents must be a list of a list
+    """
+
+    # First sort the content of the array first
+    lst.sort(key=itemgetter(0))
```
[the remark this answers is not in the document]
Comment (anhtran-ccpo) [reply atat/utils/utilities.py]: I removed this key sorting and resolved to just using sort()

File: atat/utils/utilities.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    for item in lst:
+        # Sort the list of items
+        item.sort()
+
+    return lst
```
[the remark this answers is not in the document]
Comment (anhtran-ccpo) [reply atat/utils/utilities.py]: I removed this as the lst.sort() was already handling this so it was no longer needed. I believe I had this so i could do sort_nested(arr) == sort_nested(arr2).

File: atat/utils/utilities.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    """
+    This method is for sorting a list of lists. This as it is will only work with one depth or rather a matrix
+    The contents must be a list of a list
+    """
+
+    # First sort the content of the array first
+    lst.sort(key=itemgetter(0))
+    for item in lst:
+        # Sort the list of items
+        item.sort()
```
[the remark this answers is not in the document]
Comment (anhtran-ccpo) [reply atat/utils/utilities.py]: Updated to useful more than sorting a 2D matrix array but must have the same depth.

**Comment by anhtran-ccpo** [comment]
Yeah let me know if there is anything else

**Comment by kylelaker-ccpo** [comment]
Thanks for iterating on this! The code looks great but I want to take some time to do a review of the code base and recent failures to ensure we're not missing other nondeterministic list comparisons.

**Pull request merged**